src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -937,7 +941,26 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMCluster> const& m)
         overlay_.resourceManager().importConsumers (name_, gossip);
     }
 
-    app_.getFeeTrack().setClusterFee(app_.getUNL().getClusterFee());
+    // Calculate the cluster fee:
+    auto const thresh = app_.timeKeeper().now().time_since_epoch().count() - 90;
+    std::uint32_t clusterFee = 0;
+
+    std::vector<std::uint32_t> fees;
+
+    app_.cluster().for_each(
+        [&fees,thresh](ClusterNode const& status)
+        {
+            if (status.getReportTime() >= thresh)
+                fees.push_back (status.getLoadFee ());
+        });
+
+    if (!fees.empty())
+    {
+        std::sort (fees.begin(), fees.end());
+        clusterFee = fees[fees.size() / 2];
```

Comment:
If we're only interested in `fees[fees.size()/2]` we could use `std::nth_element` instead